src/components/hoc/LoadingIf.jsx

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+import React from 'react';
+
+const UPDATE_MS = 500;
+const MAX_DOTS = 3;
+
+var LoadingIf = (Wrapped, loadingCondition) => class LoadingIf extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            dots: '',
+            loading: loadingCondition(props)
+        }
+    }
+
+    componentDidMount() {
+        this.interval = setInterval(() => {
+            let currentDots = this.state.dots + '.';
```

Comment:
No offense, but since you're already using bootstrap for your prototype (which we should remove anyways), wasn't there any spinning icon support? Like this: http://fontawesome.io/examples/#animated